diff --git a/client/components/editor/editor-modal-drawio.vue b/client/components/editor/editor-modal-drawio.vue
index 9328c45e52..6220a1891e 100644
--- a/client/components/editor/editor-modal-drawio.vue
+++ b/client/components/editor/editor-modal-drawio.vue
@@ -2,14 +2,13 @@
   v-card.editor-modal-drawio.animated.fadeIn(flat, tile)
     iframe(
       ref='drawio'
-      src='https://embed.diagrams.net/?embed=1&proto=json&spin=1&saveAndExit=1&noSaveBtn=1&noExitBtn=0'
+      :src='drawioBaseUrl'
       frameborder='0'
     )
 </template>
 
 <script>
 import { sync, get } from 'vuex-pathify'
-
 // const xmlTest = `<?xml version="1.0" encoding="UTF-8"?>
 // <mxfile version="13.4.2">
 //   <diagram id="SgbkCjxR32CZT1FvBvkp" name="Page-1">
@@ -34,7 +33,10 @@ export default {
   },
   computed: {
     editorKey: get('editor/editorKey'),
-    activeModal: sync('editor/activeModal')
+    activeModal: sync('editor/activeModal'),
+    drawioBaseUrl() {
+      return `${siteConfig.drawioBaseUrl}?embed=1&proto=json&spin=1&saveAndExit=1&noSaveBtn=1&noExitBtn=0`
+    }
   },
   methods: {
     close () {
diff --git a/server/app/data.yml b/server/app/data.yml
index 54482c99f1..7a5b8a8fcd 100644
--- a/server/app/data.yml
+++ b/server/app/data.yml
@@ -45,6 +45,7 @@ defaults:
     company: ''
     contentLicense: ''
     logoUrl: https://static.requarks.io/logo/wikijs-butterfly.svg
+    drawioBaseUrl: https://embed.diagrams.net/
     pageExtensions:
       - md
       - html
diff --git a/server/controllers/common.js b/server/controllers/common.js
index 3bcfcd940f..2fb0e736c3 100644
--- a/server/controllers/common.js
+++ b/server/controllers/common.js
@@ -113,6 +113,9 @@ router.get(['/e', '/e/*'], async (req, res, next) => {
   _.set(res, 'locals.siteConfig.lang', pageArgs.locale)
   _.set(res, 'locals.siteConfig.rtl', req.i18n.dir() === 'rtl')
 
+  // -> Set Draw.IO url for editor-modal-drawio.vue
+  _.set(res, 'locals.siteConfig.drawioBaseUrl', WIKI.config.drawioBaseUrl)
+
   // -> Check for reserved path
   if (pageHelper.isReservedPath(pageArgs.path)) {
     return next(new Error('Cannot create this page because it starts with a system reserved path.'))
