diff --git a/server/models/pages.js b/server/models/pages.js
index bb5b658512..7cd099da60 100644
--- a/server/models/pages.js
+++ b/server/models/pages.js
@@ -199,7 +199,10 @@ module.exports = class Page extends Model {
           result = frontmatterRegex.markdown.exec(raw)
           if (result[2]) {
             return {
-              ...yaml.safeLoad(result[2]),
+              ...yaml.safeLoad(result[2], {
+                json: true,
+                schema: yaml.CORE_SCHEMA
+              }),
               content: result[3]
             }
           } else {
